File: sources/Team.cpp

#include "../includes/Driver.hpp"
#include "../includes/Team.hpp"
#include "../includes/Tile.hpp"
 
Team::Team(const std::string& name, const std::string& driver1Name, const std::string& driver2Name, TireType tire, Weather weather)
    : _Name(name),
      _Driver1(driver1Name, name, tire, weather),
      _Driver2(driver2Name, name, tire, weather) {}

std::string Team::getTeamName() const {
    return _Name;
}

Driver& Team::getDriver1() {
    return _Driver1;
}

Driver& Team::getDriver2() {
    return _Driver2;
}

// void Team::resetTeamStats() {
//     _Driver1.resetStats();
//     _Driver2.resetStats();
// }

File: sources/Team.ts

import { Driver } from "./Driver.ts";
import { TireType, Weather } from "./Tire.ts";

export class Team {
  private name: string;
  private driver1: Driver;
  private driver2: Driver;

  constructor(name: string, driver1Name: string, driver2Name: string, tire: TireType, weather: Weather) {
    this.name = name;
    this.driver1 = new Driver(driver1Name, name, tire, weather);
    this.driver2 = new Driver(driver2Name, name, tire, weather);
  }

  getTeamName(): string {
    return this.name;
  }

  getDriver1(): Driver {
    return this.driver1;
  }

  getDriver2(): Driver {
    return this.driver2;
  }
}
